packages/language-service/src/locate_symbol.ts

```diff
@@ -148,18 +148,57 @@ function findAttribute(info: AstResult, position: number): Attribute|undefined {
   return path.first(Attribute);
 }
 
+function findParentOfBinding(
+    ast: TemplateAst[], binding: BoundDirectivePropertyAst, position: number) {
```

Comment:
```suggestion
    ast: TemplateAst[], binding: BoundDirectivePropertyAst, position: number): DirectiveAst|undefined {
```

Just to make it easier when the functions. I think this will allow you to get rid of the typing on 153 as well.

Reply:
You can't get rid of it. If I remove the type of `res`, the typescript will regard it as `any` which is compatible with ` DirectiveAst|undefined`. No need to change here.

Reply:
If you specify the return type, TypeScript will automatically infer the type of `res`. No need to change the type of `res`, but can you add the return type on the function definition itself, since this is a large function?